### ISSUE TITLE
Enable when playback is detected via MPRIS

### ISSUE THREAD

**Issue by carlocastoldi** [eonpatapon/gnome-shell-extension-caffeine#217]
MPRIS is a D-Bus API which aims to provide a common programmatic API for controlling media players.
In our case, it is interesting because it provides a mechanism for querying compliant media players, including music players and browsers.

Using the MPRIS API to determine if there currently is any media playing sounds like good option for this extension!

**Comment by eonpatapon** [comment]
Not sure to understand what you are proposing ?

I don't see how MPRIS/music players would interact with Caffeine

**Comment by eonpatapon** [comment]
Ah you mean maybe an option to enable Caffeine when some player is 'play' status ?

**Comment by carlocastoldi** [comment]
>Ah you mean maybe an option to enable Caffeine when some player is 'play' status ?
yes, correct! Similar to the option to enable caffeine for fullscreen apps

**Comment by snoopdouglas** [comment]
This feature would be incredibly useful to me, given that I use HDMI audio - so screen off = audio off too.

I've never done any work on GNOME extensions but would be happy to start work on a PR for this. Any pointers would be appreciated.

**Comment by eonpatapon** [comment]
Excerpts from snoopdouglas's message of February 13, 2024 9:49 am:
> This feature would be incredibly useful to me, given that I use HDMI audio - so screen off = audio off too.
> 
> I've never done any work on GNOME extensions but would be happy to start work on a PR for this. Any pointers would be appreciated.
> 

MPRIS status is exposed via dbus, so the extension would need to track via DBUS if some player is in playing state.

I think you can look for media player extensions and see how they use the MPRIS Dbus API.

Not the simplest change to do, will add quite some code to the extension.

**Comment by carlocastoldi** [comment]
I implemented MPRIS trigger in [`caffeine-ng`](https://codeberg.org/WhyNotHugo/caffeine-ng). You might want to take a hint on the dbus calls I used. It can be useful: https://codeberg.org/WhyNotHugo/caffeine-ng/src/commit/1c638ba251124adb7e4812b717735f4c35fc3784/caffeine/triggers.py#L248

The PRs are [!127](https://codeberg.org/WhyNotHugo/caffeine-ng/pulls/129) and [!146](https://codeberg.org/WhyNotHugo/caffeine-ng/pulls/146)
